src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -126,96 +125,138 @@ public async Task ExecuteAsync(ReindexJobRecord reindexJobRecord, WeakETag weakE
                     _reindexJobRecord.Count = countOnlyResults.TotalCount.Value;
 
                     // Query first batch of resources
-                    await ProcessQueryAsync(queryStatus, cancellationToken);
+                    await ProcessQueryAsync(queryStatus, jobLock, cancellationToken);
                 }
-                else
+
+                var queryTasks = new ConcurrentBag<Task>();
```

Comment:
Do we need `queryTasks` to be a ConcurrentBag? It seems like a List should be enough for the current use case.

Reply:
This is true, reverted to a List.